src/service/ratelimit/rule.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2025 OpenObserve Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::future::Future;
+
+use anyhow::Context;
+use infra::table::ratelimit::RatelimitRule;
+use o2_enterprise::enterprise::super_cluster::queue_item::ratelimit::{
+    ratelimit_rule_delete, ratelimit_rule_put, ratelimit_rule_update, SUPER_CLUSTER_TOPIC_RATELIMIT,
```

Comment:
`SUPER_CLUSTER_TOPIC_RATELIMIT` looks just a message prefix, it is not topic, `topic` is equal to queue name.